src/python/test/test_dxclient.py

```diff
@@ -10254,7 +10254,191 @@ def test_update_app(self):
         self.assertEqual(app_2['name'], app_spec_2['name'])
         self.assertEqual(app_2['version'], "0.0.2")
 
+class TestDXArchive(DXTestCase):
+    @classmethod
+    def setUpClass(cls):
+        # setup two projects
+        cls.proj_archive_name = "dx_test_archive"
+        cls.proj_unarchive_name = "dx_test_unarchive"
+        cls.usr = dxpy.whoami()
+        cls.bill_to = dxpy.api.user_describe(cls.usr)['billTo']
+        cls.is_admin = True if dxpy.api.org_describe(cls.bill_to)['level'] == 'ADMIN' else False
+
+        cls.rootdir = '/test-folder'
+        cls.proj_archive_id = dxpy.api.project_new({'name': cls.proj_archive_name, 'billTo': cls.bill_to})['id']
+        cls.proj_unarchive_id = dxpy.api.project_new({'name': cls.proj_unarchive_name, 'billTo': cls.bill_to})['id']
+        cls.counter = 1
+        
+        create_folder_in_project(cls.proj_archive_id, cls.rootdir)
+        create_folder_in_project(cls.proj_unarchive_id, cls.rootdir)
+        
+    @classmethod
+    def tearDownClass(cls):
+        dxpy.api.project_remove_folder(cls.proj_archive_id, {"folder": cls.rootdir, "recurse":True})
+        dxpy.api.project_remove_folder(cls.proj_unarchive_id, {"folder": cls.rootdir,"recurse":True})
+        rm_project(cls.proj_archive_id)
+        rm_project(cls.proj_unarchive_id)
 
+    @classmethod
+    def gen_uniq_fname(cls):
+        cls.counter += 1
+        return "file_{}".format(cls.counter)
+  
+    def test_archive_files(self):
```

Comment:
May also want to add a test case that mixes in files and folders to ensure there's no unexpected behavior when you have a mix of types.

Reply:
currently user can supply one of
* a list of file paths
* a single folder